From e8a7bf587946a6ca6ebb168147e1461a8f029bbe Mon Sep 17 00:00:00 2001
From: Mei Xie <quic_meixie@quicinc.com>
Date: Thu, 10 Jul 2025 17:54:16 +0800
Subject: [PATCH 14/28] PENDING: net: stmmac: change mapping between
 VDMA/TC/PDMA for CBS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the VDMA–TC–PDMA configuration to a 1:1 mapping to align with
software TC mapping and ensure CBS functionality.

Change-Id: I9ce3d93b83bcdcad21f92834c75be9dd8cf34968
Signed-off-by: Mei Xie <quic_meixie@quicinc.com>
---
 .../stmicro/stmmac/dwmac-qcom-ethqos.c        | 32 +++++++++----------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
index 60fdb71ac38b3..0a6a007f40ddc 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-qcom-ethqos.c
@@ -838,30 +838,30 @@ static void qcom_ethqos_hdma_cfg(struct plat_stmmacenet_data *plat)
 
 	plat->dma_cfg->tx_pdma_custom_map = true;
 	plat->dma_cfg->tx_pdma_map[0] = 0;
-	plat->dma_cfg->tx_pdma_map[1] = 0;
-	plat->dma_cfg->tx_pdma_map[2] = 0;
-	plat->dma_cfg->tx_pdma_map[3] = 0;
-	plat->dma_cfg->tx_pdma_map[4] = 5;
+	plat->dma_cfg->tx_pdma_map[1] = 1;
+	plat->dma_cfg->tx_pdma_map[2] = 2;
+	plat->dma_cfg->tx_pdma_map[3] = 3;
+	plat->dma_cfg->tx_pdma_map[4] = 4;
 	plat->dma_cfg->tx_pdma_map[5] = 5;
 	plat->dma_cfg->tx_pdma_map[6] = 5;
-	plat->dma_cfg->tx_pdma_map[7] = 2;
-	plat->dma_cfg->tx_pdma_map[8] = 3;
-	plat->dma_cfg->tx_pdma_map[9] = 4;
-	plat->dma_cfg->tx_pdma_map[10] = 6;
+	plat->dma_cfg->tx_pdma_map[7] = 5;
+	plat->dma_cfg->tx_pdma_map[8] = 6;
+	plat->dma_cfg->tx_pdma_map[9] = 6;
+	plat->dma_cfg->tx_pdma_map[10] = 7;
 	plat->dma_cfg->tx_pdma_map[11] = 7;
 
 	plat->dma_cfg->rx_pdma_custom_map = true;
 	plat->dma_cfg->rx_pdma_map[0] = 0;
-	plat->dma_cfg->rx_pdma_map[1] = 0;
-	plat->dma_cfg->rx_pdma_map[2] = 0;
-	plat->dma_cfg->rx_pdma_map[3] = 0;
-	plat->dma_cfg->rx_pdma_map[4] = 5;
+	plat->dma_cfg->rx_pdma_map[1] = 1;
+	plat->dma_cfg->rx_pdma_map[2] = 2;
+	plat->dma_cfg->rx_pdma_map[3] = 3;
+	plat->dma_cfg->rx_pdma_map[4] = 4;
 	plat->dma_cfg->rx_pdma_map[5] = 5;
 	plat->dma_cfg->rx_pdma_map[6] = 5;
-	plat->dma_cfg->rx_pdma_map[7] = 2;
-	plat->dma_cfg->rx_pdma_map[8] = 3;
-	plat->dma_cfg->rx_pdma_map[9] = 4;
-	plat->dma_cfg->rx_pdma_map[10] = 6;
+	plat->dma_cfg->rx_pdma_map[7] = 5;
+	plat->dma_cfg->rx_pdma_map[8] = 6;
+	plat->dma_cfg->rx_pdma_map[9] = 6;
+	plat->dma_cfg->rx_pdma_map[10] = 7;
 	plat->dma_cfg->rx_pdma_map[11] = 7;
 }
 
